pkg/pillar/devicenetwork/addrchange.go

```diff
@@ -167,6 +167,13 @@ func HandleAddressChange(ctx *DeviceNetworkContext) {
 		status, _ := MakeDeviceNetworkStatus(*ctx.DevicePortConfig,
```

Comment:
There are 5 or 6 places we call MakeDeviceNetworkStatus, and you've only added code for one of them.

Furthermore, we receate/update DevcieNetworkStatus and the PendingDNS in many places, which means that we loose the CloudReachable setting (meaning it defaults to false hence send will not even try to use it).
This clearly isn't anywhere near robust enough.

One could make it be somewhat usable by having a CloudUnreachable which is only set when it is known to be unreachable. But still begs the question of "why?". The configuration with incorrectly configured management ports is incorrect and needs to be fixed. We should spend effort on the UI to tell the user it is broken and how to fix the config. Not pretending that everything is fine by adding this complexity.

The cloud is correct in saying "Suspect" because the device configuration is highly suspect!

Reply:
@eriknordmark There are only couple of places where copying CloudReachable flag needs be done into new DeviceNetworkStatus from old status. The other cases where we call MakeDeviceNetworkStatus are in the paths where we are in the process of evaluating device connectivity to cloud. CloudReachable flag should get updated according during the evaluation process.

With the cloud Reachable flag we can tell cloud that there are specific interface configured to be management but does not have the right configuration/connectivity.

Reply:
But the default of cloudreachable=false when you create a new object is completely broken.
You need to default to treating it as reachable otherwise zedagent and friends will default to talk to nobody.
Hence by suggestion of a cloudUNreachable boolean.

Reply:
During the device boot, NIM would have figured out the connectivity through different uplink ports and updated their CloudReachable flag appropriately. I agree with what your said during the transient system adapter configuration changes. Sure, I'll make the flag CloudUnReachable.

Reply:
Hmm.. cloudreachable=false  --> Why not set this to True by default? The variable name CloudReachable is much more readable than "CloudUnreachable"

One way to force users to use the Struct with default values set is:
https://stackoverflow.com/questions/37135193/how-to-set-default-values-in-go-structs